cpp/cmake/thirdparty/CUDF_GetArrow.cmake

```diff
@@ -0,0 +1,123 @@
+#=============================================================================
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+function(find_and_configure_arrow VERSION BUILD_STATIC)
+
+    set(ARROW_BUILD_SHARED ON)
```

Comment:
When would we want to build a shared library as opposed to just finding it? I assume if we build a shared library we don't have a mechanism for installing it?

Reply:
If we're building shared, and it's already installed on the system/by conda, then finding it should be fine. Otherwise, CPM will build it.

Reply:
I guess my point is if we're building libcudf as a shared library it would required the Arrow shared library to be installed in order to function properly. If we build the Arrow shared libraries here they won't be installed and we'll end up in a broken state in installation of libcudf. From my perspective we should allow 3 paths:
- Find shared libraries for Arrow and link using them
- Find static libraries for Arrow and link using them
- Build static libraries for Arrow and link using them